### PR TITLE
Add support for `min_const_generics` constants

### PR DESCRIPTION
**Pull request by varkor**
Add demangling for `bool`, signed integers, and `char`.

cc @eddyb

### CODE REVIEW

File: src/v0.rs [hunk 1]
```diff
@@ -1028,6 +1090,22 @@ mod tests {
             "INtC8arrayvec8ArrayVechKj7b_E",
             "arrayvec::ArrayVec<u8, 123>"
         );
+        t_nohash!(
+            "_RMCs4fqI2P2rA04_13const_genericINtB0_8UnsignedKhb_E",
+            "<const_generic::Unsigned<11>>"
+        );
+        t_nohash!(
+            "_RMCs4fqI2P2rA04_13const_genericINtB0_6SignedKanb_E",
+            "<const_generic::Signed<-11>>"
```
Comment (alexcrichton) [inline src/v0.rs:1099]: For completion's sake mind adding a test for a positive signed number as well?

(also a `false` and a different `char` value too?)

Comment (varkor) [reply src/v0.rs:1103]: Done!

**Comment by alexcrichton** [comment]
👍 thanks!

Would you like a publish for this as well?

**Comment by varkor** [comment]
Actually, there's one other change @eddyb suggested (I forgot to include it in this PR), so I'll make that first: thanks!

File: src/v0.rs [hunk 2]
```diff
@@ -972,6 +999,41 @@ impl<'a, 'b, 's> Printer<'a, 'b, 's> {
...
+        if hex.len() > 8 {
+            invalid!(self);
+        }
+
+        let mut v = 0;
+        for c in hex.chars() {
+            v = (v << 4) | (c.to_digit(16).unwrap() as u32);
+        }
+        if let Some(c) = char::from_u32(v) {
+            write!(self.out, "'{}'", c)
```
Comment (eddyb) [inline src/v0.rs:1032]: This should use `"{:?}"` so that you get e.g. `'\n'` instead of literally a newline in the output etc.

**Pull request merged**